sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1353,9 +1353,20 @@ void simd_obj_impl<T, N, T1, SFINAE>::copy_to(
       if constexpr (RemN == 1) {
         Addr[NumChunks * ChunkSize] = Tmp[NumChunks * ChunkSize];
       } else if constexpr (RemN == 8 || RemN == 16) {
-        simd<uint32_t, RemN> Offsets(0u, sizeof(T));
-        scatter<UT, RemN>(Addr + (NumChunks * ChunkSize), Offsets,
-                          Tmp.template select<RemN, 1>(NumChunks * ChunkSize));
+        if constexpr (sizeof(T) == 1 && RemN == 16) {
+          simd<int32_t, 8> Vals;
+          Vals.template select<4, 1>() =
+              Tmp.template bit_cast_view<int32_t>().template select<4, 1>(
+                  NumChunks * ChunkSize);
+          block_store<int32_t, 8>(reinterpret_cast<int32_t *>(Addr) +
```

Comment:
How does this work? Vals has 8 int elements where the low 4 hold some useful values, while the upper 4 keep garbage, which is block_store'd to memory. 
Looks like it writes to memory that is not supposed to be written, right?